lib/engine/game/g_1858/company.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require_relative '../../company'
+require_relative '../../operator'
+
+module Engine
+  module Game
+    module G1858
+      # Private companies in 1858 are a hybrid with elements of other games'
```

Comment:
ATM I'm suspicious of trying to treat the 1858 privates as both private companies and minor companies, it seems like it might simplify a few things to make them minors that happen to also pay. I'll tinker with that and hopefully have more firm thoughts later on.

Reply:
agree, i think this PR should be redone companies as minors instead of this weird hybrid

Reply:
The hybrid company/minor class has been removed.